speedify.py

```diff
@@ -424,11 +594,348 @@ def safebrowsing_stats():
+        )
+
+    :param operation: The operation to perform. One of:
+        "add"
+        "rem"
+        "set"
+    :type operation: str
+    :param ipv6_addrs: The ipv6 address(es) to bypass. Example:
+        "2001:db8:1234:ffff:ffff:ffff:ffff:0f0f 2001:db8:1234:ffff:ffff:ffff:ffff:ffff"
+        "2001:db8:1234:ffff:ffff:ffff:ffff:ffff"
+    :type ipv6_addrs: str
+    """
+    valid_operations = ["add", "rem", "set"]
+    if operation in valid_operations:
+        return _run_speedify_cmd(["streamingbypass", "ipv6", operation, ipv6_addrs])
+    else:
+        raise ValueError("Invalid operation: " + operation)
+
+
+@exception_wrapper("Failed to set streaming bypass")
+def streamingbypass_ports(operation: str, ports: str):
+    """
+    streamingbypass_ports(operation, ports)
+
+    Add, remove, or set the streaming bypass for some ports.
+
+    Example:
+        streamingbypass_ports("rem", "9999/tcp")
+
+    :param operation: The operation to perform. One of:
+        "add"
+        "rem"
+        "set"
+    :type operation: str
+    :param ports: The ports to bypass. Must be of one of these forms:
+        "<port>/<proto>"
+        "<port begin>-<port end>/<proto>"
+    :type ports: str
+    """
+    valid_operations = ["add", "rem", "set"]
+    if operation in valid_operations:
+        return _run_speedify_cmd(["streamingbypass", "ports", operation, ports])
+    else:
+        raise ValueError("Invalid operation: " + operation)
+
+
+@exception_wrapper("Failed to set streaming bypass")
+def streamingbypass_service(service_name: str, is_on: bool):
+    """
+    streamingbypass_service(service_name, is_on)
+
+    Set the streaming bypass, on or off, for some pre-defined service.
+
+    Example:
+        streamingbypass_service("Netflix", True)
+
+    :param service_name: The service to modify. One of:
+        "Netflix"
+        "Disney+"
+        "HBO"
+        "Hulu"
+        "Peacock"
+        "Amazon Prime"
+        "Youtube TV"
+        "Ring"
+        "VoLTE"
+        "Reliance Jio"
+        "Microsoft Your Phone"
+        "Spectrum TV & Mobile"
+        "Showtime"
+        "Visual Voice Mail"
+        "Android Auto"
+        "Tubi"
+        "Hotstar"
+        "RCS Messaging"
+        "Ubisoft Connect"
+        "Apple Updates"
+    :type service_name: str
+    :param is_on: Whether to bypass the service... or not.
+    :type is_on: bool
+    """
+    valid_service_names = [
+        "Netflix",
+        "Disney+",
+        "HBO",
+        "Hulu",
+        "Peacock",
+        "Amazon Prime",
+        "Youtube TV",
+        "Ring",
+        "VoLTE",
+        "Reliance Jio",
+        "Microsoft Your Phone",
+        "Spectrum TV & Mobile",
+        "Showtime",
+        "Visual Voice Mail",
+        "Android Auto",
+        "Tubi",
+        "Hotstar",
+        "RCS Messaging",
+        "Ubisoft Connect",
+        "Apple Updates",
+    ]
+    if service_name in valid_service_names:
```

Comment:
I don't think we need the verification on this side. This is another list that needs to be updated when we add new services and can easily get out of date. I think you can let the daemon handle checking whether the argument passed in is a valid service.

Reply:
`speedify_cli show streamingbypass | grep title | sed -E 's/.*: (.*)/\1,/g'`

Which is ultra easy to automate

Reply:
I still don't think you need this verification here. Let the daemon be the source of truth. A user could be running this library against an older or newer version that does or does not have a service in this list.

Reply:
On the other hand, I agree that doing error checking in the python script doesn't seem necessary. I can remove this.

Reply:
Saw your comment after refresh (and after posting my comment no. 2)